subworkflows/local/kofamscan.nf

```diff
@@ -0,0 +1,28 @@
+//
+// Run KOFAMSCAN on protein fasta from orf_caller output
+//
+
+include { KOFAMSCAN     } from '../../modules/local/eukulele/kofamscan'
+include { DOWNLOAD_KODB } from '../../modules/local/eukulele/download_kofamscan_db'
+
+workflow SUB_KOFAMSCAN {
+
+    take:
+        kofamscan // Channel: val(meta), path(fasta), val(database), path(ko_list), path(koprofiles)
```

Comment:
What are all those params? Aren't `database`, `ko_list` and `koprofiles` paths to data in the database? I would guess that only the first is needed.

Reply:
As far as I understood about how kofamscan works, I need to specify the path for the profiles and the ko_list, so I will need two different path. Val(database) is just the name for the database (always kofamscan in this case)

Reply:
Correct. I suppose my point is that that might not be necessary to know for the caller of the subworkflow, i.e. default paths starting from `database` is maybe possible. But they should perhaps be possible to specify.
Another, but related, thing is that I think the nf-core people seem to want separate channels for the different pieces, each with a `meta` if you might need to be able to merge them (which I don't think is the case here so you could skip `meta` for all except the first). That would also make it slightly easier to specify which are non-mandatory.

Reply:
BTW, [here's](https://nfcore.slack.com/archives/C043UU89KKQ/p1678700182885289) a discussion regarding whether `meta` is compulsory or not (it's not). It's about modules, but I think the same principles apply to subworkflows: Add a `meta` to a channel if it makes sense, in this case it probably doesn't.